smalltalksrc/Slang/TParseNode.class.st

```diff
@@ -283,6 +283,15 @@ TParseNode >> isNonNullCCode [
 	^false
 ]
 
+{ #category : 'testing' }
+TParseNode >> isNotExpression [
+
+	^ parent isNil or: [
+		  parent isTMethod or: [
+			  parent isSend not and: [
+				  parent isReturn not and: [ parent isAssignment not ] ] ] ]
+]
```

Comment:
I don't understand completely this:

1. The method is _negated_ (is**Not**Expression). It is more difficult to follow a negation than an affirmation. Would be nice to have an affirmative method and do the negation in the use: `self isExpression not`.
2. This definition has sense for all `TParseNode`? Or only for `TReturnNode` (who call this method).
3. I don't follow the condition anyway:
  - Why if the paren is nil is not an expression?
  - I understand that the only _non-expression_ nodes are returns, assignments, (maybe if / case ?). Why is not asking that and is always asking for the parent?

Reply:
Let's open an issue for this